public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -1339,7 +1340,23 @@ export class DashboardModel implements TimeModel {
     return this.originalDashboard;
   }
 
-  hasAngularPlugins(): boolean {
+  async hasAngularPlugins(): Promise<boolean> {
+    // Ensure all panel plugins are loaded before checking if they are Angular plugins
+    await Promise.all(
+      this.panels.map(async (panel) => {
+        if (panel.plugin) {
+          return;
+        }
+        try {
+          const plugin = await dispatch(loadPanelPlugin(panel.type));
```

Comment:
Now sure how big a deal this is (as from what I can see this isn't render blocking) but would like to point out the current loading mechanism only loads panel plugins when the panel comes into view (i.e. as a user scrolls). This change means all panel plugins for a dashboard are loaded up front now.